### PR TITLE
Make filter mutator API consistent with search mutator API.

### PR DESCRIPTION
**Pull request by askvortsov1**
This is inline with the docblock for the Filter extender, and is much more sensible.

### CODE REVIEW

**Review by SychO9 — APPROVED** [review]

**Comment by clarkwinkelmann** [comment]
I have not well followed what was going on with this. Nothing was using the feature that is modified by this PR?

**Comment by askvortsov1** [comment]
> I have not well followed what was going on with this. Nothing was using the feature that is modified by this PR?

Correct, this was all introduced in this release, and not even the bundled extensions have been updated for it yet.

**Review by SychO9 — APPROVED** [review]

File: src/Filter/AbstractFilterer.php [hunk 1]
```diff
@@ -77,7 +78,7 @@ public function filter(QueryCriteria $criteria, int $limit = null, int $offset =
         // END DEPRECATED BC LAYER
 
         foreach ($this->filterMutators as $mutator) {
-            $mutator($query, $actor, $criteria->query, $criteria->sort);
```
Comment (KyrneDev) [inline src/Filter/AbstractFilterer.php:80]: So is actor not available anymore?

**Review by KyrneDev — CHANGES_REQUESTED** [review]

File: src/Filter/AbstractFilterer.php (hunk 1 above)
Comment (askvortsov1) [reply src/Filter/AbstractFilterer.php:80]: It is: filter state is a wrapper around the query builder and the actor.

**Review by KyrneDev — APPROVED** [review]

**Review by tankerkiller125 — APPROVED** [review]

**Pull request merged**